ecllibrary/teststd/uni/TestExcludeNthWord.ecl

```diff
@@ -0,0 +1,65 @@
+     //Check action on a string with no entries.
+     EXPORT Test01 := ASSERT(Uni.ExcludeNthWord('',0)+'!' = '!', CONST);
+     EXPORT Test02 := ASSERT(Uni.ExcludeNthWord('',1)+'!' = '!', CONST);
+     EXPORT Test03 := ASSERT(Uni.ExcludeNthWord('',-1)+'!' = '!', CONST);
+     EXPORT Test04 := ASSERT(Uni.ExcludeNthWord('             ',0)+'!' = '!', CONST);
+     EXPORT Test05 := ASSERT(Uni.ExcludeNthWord('             ',1)+'!' = '!', CONST);
+     EXPORT Test06 := ASSERT(Uni.ExcludeNthWord('             ',-1)+'!' = '!', CONST);
+     //Check action on a string containing a single word - with various whitespace
+     EXPORT Test07 := ASSERT(Uni.ExcludeNthWord('x',0)+'!' = 'x!');
+     EXPORT Test08 := ASSERT(Uni.ExcludeNthWord('x',1)+'!' = '!');
+     EXPORT Test09 := ASSERT(Uni.ExcludeNthWord('x',2)+'!' = 'x!');
+     EXPORT Test10 := ASSERT(Uni.ExcludeNthWord('x',3)+'!' = 'x!');
+     EXPORT Test11 := ASSERT(Uni.ExcludeNthWord(' x',1)+'!' = '!');
+     EXPORT Test12 := ASSERT(Uni.ExcludeNthWord('x ',1)+'!' = '!');
+     EXPORT Test13 := ASSERT(Uni.ExcludeNthWord(' x',2)+'!' = ' x!');
+     EXPORT Test14 := ASSERT(Uni.ExcludeNthWord(' x ',1)+'!' = '!');
+     //Check action on a string containg multiple words - with various whitespace combinations.
+     EXPORT Test15 := ASSERT(Uni.ExcludeNthWord(' abc def ', 1)+'!' = 'def !');
+     EXPORT Test16 := ASSERT(Uni.ExcludeNthWord(' abc def ', 2)+'!' = ' abc !');
+     EXPORT Test17 := ASSERT(Uni.ExcludeNthWord('  a b c   def    ',0)+'!' = '  a b c   def    !');
+     EXPORT Test18 := ASSERT(Uni.ExcludeNthWord('  a b c   def    ',1)+'!' = 'b c   def    !');
+     EXPORT Test19 := ASSERT(Uni.ExcludeNthWord('  a b c   def    ',2)+'!' = '  a c   def    !');
+     EXPORT Test20 := ASSERT(Uni.ExcludeNthWord('  a b c   def    ',3)+'!' = '  a b def    !');
+     EXPORT Test21 := ASSERT(Uni.ExcludeNthWord('  a b c   def    ',4)+'!' = '  a b c   !');
+     EXPORT Test22 := ASSERT(Uni.ExcludeNthWord('  a b c   def    ',5)+'!' = '  a b c   def    !');
+     EXPORT Test23 := ASSERT(Uni.ExcludeNthWord(' ,,,, ',1)+'!' = '!');
+     //Test other space characters (< 0x20)
+     EXPORT Test24 := ASSERT(Uni.ExcludeNthWord('  a b\nc \t  def    ',2)+'!' = '  a c \t  def    !');
+     EXPORT Test25 := ASSERT(Uni.ExcludeNthWord('  a b\nc \t  def    ',3)+'!' = '  a b\ndef    !');
+     //Test latin diacritical marks
+     EXPORT Test26 := ASSERT(Uni.ExcludeNthWord(U'À à',2)+U'!' = U'À !');
+     EXPORT Test27 := ASSERT(Uni.ExcludeNthWord(U'ä̰́ Ä̰́',2)+U'!' = U'ä̰́ !');
+     //Test Chinese. "I am a computer" --> "I am"
+     EXPORT Test28 := ASSERT(Uni.ExcludeNthWord(U'我是電腦',2)+U'!' = U'我是!');
+     /*
+     //arabic lettering with accent marks. Bidirectional.
+     EXPORT Test29 := ASSERT(Uni.ExcludeNthWord(U'حركات إعجام', 2)+U'!' = U' حركات!');
+     //hebrew lettering with accent marks (called pointing). Bidirectional.
+     EXPORT Test30 := ASSERT(Uni.ExcludeNthWord(U'קָמָץ שִׁי״ן',2)+U'!' = U' קָמָץ!');
+     */
```

Comment:
Are there any locales that make a difference to the word rules?  As far as I can see the special rules for thai/chinese are used whatever the locale.  If there are no examples it would be worth adding a comment to that effect here.